### DISCUSSION TITLE
Rate Limit per user

### DISCUSSION THREAD

**Discussion by myasul** [rabbitmq/rabbitmq-server#7945] [Questions]
We plan to create a distinct user per IoT device that connects to our RabbitMQ server to improve our security. Is it possible to add a rate limit per user so that if someone gets hold of one of the user credentials, they cannot pump messages into the queue in a very short span of time (DDOS attack)?

**Comment by michaelklishin** [comment]
RabbitMQ does not support any kind of rate limits in terms of bytes/second. You can use HAproxy to set up [rate limits per connection](https://www.haproxy.com/blog/four-examples-of-haproxy-rate-limiting/#:~:text=Rate%20limiting%20in%20HAProxy%20stops,of%20keeping%20resource%20usage%20fair.).

On top of that, [amqpprox](https://github.com/bloomberg/amqpprox) supports rate limiting per virtual host, which do not exist in MQTT but with RabbitMQ specifically, you can use [a couple of features and conventions](https://rabbitmq.com/mqtt.html#virtual-hosts). Maybe that would fit your needs since you'd be able to group devices that way.

**Reply by myasul** [reply]
Will definitely look into this. Thank you very much for the quick response, @michaelklishin !
